compiler-core/src/language_server/tests/action.rs

```diff
@@ -91,13 +92,13 @@ fn apply_code_edit(
             panic!("Unknown url {}", change_url)
         }
         for edit in change {
-            let start =
-                line_numbers.byte_index(edit.range.start.line, edit.range.start.character) - offset;
-            let end =
-                line_numbers.byte_index(edit.range.end.line, edit.range.end.character) - offset;
+            let start = line_numbers.byte_index(edit.range.start.line, edit.range.start.character)
+                as i32
+                - offset;
+            let end = line_numbers.byte_index(edit.range.end.line, edit.range.end.character) as i32
+                - offset;
             let range = (start as usize)..(end as usize);
-            offset += end - start;
-            offset -= edit.new_text.len() as u32;
+            offset += end - start - edit.new_text.len() as i32;
```

Comment:
Revert this please, this old version has it has a bug in it 🙏

Reply:
The way it was before with `offset` being a `u32`, the subtraction on line 100 would panic with an integer underflow if the code action added more characters than it removed. I'm not sure how else to solve this. What exactly is the bug?

Reply:
It would not apply the text edit correctly. The underflow likely means you have a bug and are trying to remove more content than exists.